Mathlib/Algebra/Group/Equiv/Basic.lean

```diff
@@ -80,10 +81,61 @@ def arrowCongr {M N P Q : Type*} [Mul P] [Mul Q] (f : M ≃ N) (g : P ≃* Q) :
   right_inv k := by ext; simp
   map_mul' h k := by ext; simp
 
+section monoidHomCongrEquiv
+variable [MulOneClass M] [MulOneClass M₁] [MulOneClass M₂] [Monoid N] [Monoid N₁] [Monoid N₂]
+
+/-- The equivalence `(M₁ →* N) ≃ (M₂ →* N)` obtained by postcomposition with
+a multiplicative equivalence `e : M₁ ≃* M₂`. -/
+@[to_additive (attr := simps)
+"The equivalence `(M₁ →+ N) ≃ (M₂ →+ N)` obtained by postcomposition with
+an additive equivalence `e : M₁ ≃+ M₂`."]
+def monoidHomCongrLeftEquiv (e : M₁ ≃* M₂) : (M₁ →* N) ≃ (M₂ →* N) where
```

Comment:
For each of these, can you add the theorem that `congr refl = refl` and `(congr e).symm = congr e.symm`?

The trans one would be good too, but I know adding variables can be annoying, so a TODO would be ok for those.